# Decision Tree "Rough Draft"
##### Getting a decision tree output so we have some results to evaluate

### Imports

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


%matplotlib inline

In [4]:
df = pd.read_csv('Event Recordings/vol1.csv')
# Display a row with a NaN value
# print(df.iloc[2249])

In [6]:
# Import our function
from DataCleaningFunction import clean_data

##### Now that we've imported our data cleaning function, we can clean our data and check if it looks correct

In [27]:
df = clean_data('Event Recordings/vol1.csv', True)
df.tail()

Timestamp    X     Y  TOUCH_MAJOR  TOUCH_MINOR  TRACKING_ID  PRESSURE  \
59545  898.462922  675   303            9            3   4294967295        31   
59552  898.621411  677  1829            9            3   4294967295        31   
59556  898.663272  677   303            9            3   4294967295        31   
59563  898.871367  677   303            9            3   4294967295        31   
59564  898.904673  677   304            9            3   4294967295        31   

       FINGER  BTN_TOUCH_HELD  BTN_TOUCH_UP      X_Speed  X_Acceleration  \
59545       0               1             0     0.000000    -3752.301976   
59552       0               1             0 -1288.659794   -51895.127010   
59556       0               1             0     0.000000        0.000000   
59563       0               1             0     0.000000        0.000000   
59564       0               1             0     0.000000        0.000000   

            Y_Speed  Y_Acceleration         Speed  Acceleration          Jerk  \
59545      0.000000    3.752302e+03      0.000000 -5.306556e+03 -7.392617e+05   
59552      0.000000   -7.245103e+06   1288.659794 -7.193208e+06 -2.856750e+08   
59556 -60821.327376   -2.422784e+06  60821.327376  2.422784e+06  1.149078e+08   
59563 -60826.763782   -2.433541e+06  60826.763782  2.433541e+06  9.796070e+07   
59564     30.024620    1.827202e+06     30.024620 -1.825399e+06 -1.278730e+08   

           Ang_V  Path_Tangent  User  
59545  97.456032      0.000000     0  
59552  63.256940      3.141593     0  
59556   0.000000     -1.570796     0  
59563   0.000000     -1.570796     0  
59564  94.325126      1.570796     0

##### *a little concerned that all 59,564 rows are coming up when we wanted it to drop NaN values*

In [28]:
# Read in volunteer 2's data
df2 = pd.read_csv('Event Recordings/vol2.csv')
df2 = clean_data('Event Recordings/vol2.csv')

# Append user 1 and user 2's dataframes
df3 = pd.concat([df, df2], ignore_index=True)

# Display the tail to ensure CORRECT_USER isn't preset
df3.tail()

Timestamp    X     Y  TOUCH_MAJOR  TOUCH_MINOR  TRACKING_ID  PRESSURE  \
52698  900.161091  311  1760           10            3         4422        27   
52699  900.170699  310  1754           10            3         4422        27   
52700  900.179196  309  1748           10            3         4422        27   
52701  900.203281  305  1733           10            3         4422        27   
52702  900.210968  303  1728           10            3         4422        27   

       FINGER  BTN_TOUCH_HELD  BTN_TOUCH_UP     X_Speed  X_Acceleration  \
52698       0               1             0 -125.250501   -15687.688002   
52699       0               1             0 -104.079933     2203.431267   
52700       0               1             0 -117.688596    -1601.584395   
52701       0               1             0 -126.055717    13683.144954   
52702       0               1             0 -260.179524   -17448.134172   

          Y_Speed  Y_Acceleration       Speed  Acceleration          Jerk  \
52698 -626.252505   -50345.919887  638.654749  51899.307122  2.667056e+09   
52699 -624.479600      184.523804  633.093519   -578.812422 -5.461919e+06   
52700 -706.131576    -9609.506372  715.871782   9742.057549  1.214649e+06   
52701 -504.222866    25159.391169  519.741034 -28002.305257  2.792795e+09   
52702 -650.448810   -19022.498176  700.554808  23522.020777  6.702787e+06   

           Ang_V  Path_Tangent  User  
52698 -24.723893     -1.768192     1  
52699   3.356253     -1.735945     1  
52700   0.000000     -1.735945     1  
52701   9.677536     -1.815775     1  
52702 -17.630768     -1.951303     1

In [29]:
df3.tail()

Timestamp    X     Y  TOUCH_MAJOR  TOUCH_MINOR  TRACKING_ID  PRESSURE  \
52698  900.161091  311  1760           10            3         4422        27   
52699  900.170699  310  1754           10            3         4422        27   
52700  900.179196  309  1748           10            3         4422        27   
52701  900.203281  305  1733           10            3         4422        27   
52702  900.210968  303  1728           10            3         4422        27   

       FINGER  BTN_TOUCH_HELD  BTN_TOUCH_UP     X_Speed  X_Acceleration  \
52698       0               1             0 -125.250501   -15687.688002   
52699       0               1             0 -104.079933     2203.431267   
52700       0               1             0 -117.688596    -1601.584395   
52701       0               1             0 -126.055717    13683.144954   
52702       0               1             0 -260.179524   -17448.134172   

          Y_Speed  Y_Acceleration       Speed  Acceleration          Jerk  \
52698 -626.252505   -50345.919887  638.654749  51899.307122  2.667056e+09   
52699 -624.479600      184.523804  633.093519   -578.812422 -5.461919e+06   
52700 -706.131576    -9609.506372  715.871782   9742.057549  1.214649e+06   
52701 -504.222866    25159.391169  519.741034 -28002.305257  2.792795e+09   
52702 -650.448810   -19022.498176  700.554808  23522.020777  6.702787e+06   

           Ang_V  Path_Tangent  User  
52698 -24.723893     -1.768192     1  
52699   3.356253     -1.735945     1  
52700   0.000000     -1.735945     1  
52701   9.677536     -1.815775     1  
52702 -17.630768     -1.951303     1

### Now, train test split
##### We need to train the data of only the correct user (70% of their data) and test the data with the remaining data (30% both correct user and incorrect users).

In [30]:
from sklearn.model_selection import train_test_split

In [32]:
# Set x data to everything but the target (of correct users)
X = df.drop('User', axis=1)

# Use the variable below to test against all users
X1 = df3.drop('User', axis=1)

In [33]:
# Do the same for y
y = df['User']

# Use the variable below to test against all users
y1 = df3['User']

# **I THINK THE PROBLEM IS HERE**

In [34]:
X_train, X_test, y_train, y_test = train_test_split(X1, y1, test_size=0.3, shuffle=True)

'''
# Data to be trained (70% of user 1's data)
X_train = train_test_split(X, y, train_size=0.7, random_state=101)
y_train = train_test_split(X, y, train_size=0.7, random_state=101)

# Data to be tested (30% of both correct and incorrect data)
X_test = train_test_split(X1, y1, test_size=0.3, random_state=101)
y_test = train_test_split(X1, y1, test_size=0.3, random_state=101)
'''

"\n# Data to be trained (70% of user 1's data)\nX_train = train_test_split(X, y, train_size=0.7, random_state=101)\ny_train = train_test_split(X, y, train_size=0.7, random_state=101)\n\n# Data to be tested (30% of both correct and incorrect data)\nX_test = train_test_split(X1, y1, test_size=0.3, random_state=101)\ny_test = train_test_split(X1, y1, test_size=0.3, random_state=101)\n"

# Implementing the Decision Tree
##### Now that we've created our train test split, we can implement the Decision Tree

In [35]:
from sklearn.tree import DecisionTreeClassifier

In [36]:
dtree = DecisionTreeClassifier()

In [37]:
dtree.fit(X_train, y_train)

DecisionTreeClassifier()

In [38]:
predictions=dtree.predict(X_test)

In [39]:
from sklearn.metrics import classification_report, confusion_matrix

In [40]:
print(confusion_matrix(y_test,predictions))
print('\n')
print(classification_report(y_test,predictions))

[[8360   63]
 [  66 7322]]


              precision    recall  f1-score   support

           0       0.99      0.99      0.99      8423
           1       0.99      0.99      0.99      7388

    accuracy                           0.99     15811
   macro avg       0.99      0.99      0.99     15811
weighted avg       0.99      0.99      0.99     15811



# Try to manually split the data then run it